control-plane/agents/src/bin/core/node/operations_helper.rs

```diff
@@ -0,0 +1,88 @@
+use agents::errors::SvcError;
+
+use stor_port::types::v0::{store::node::NodeSpec, transport::VolumeId};
+
+use crate::controller::{registry::Registry, resources::OperationGuardArc};
+
+use std::{collections::HashSet, time::SystemTime};
+
+impl OperationGuardArc<NodeSpec> {
+    /// Add the draining volume to the node spec for checking shutdown nexuses.
+    pub async fn add_node_draining_volumes(
+        &self,
+        registry: &Registry,
+        draining_volumes: HashSet<VolumeId>,
+    ) -> Result<(), SvcError> {
+        let drained_node_spec = {
+            let mut locked_node = self.lock();
+            locked_node.add_draining_volumes(draining_volumes);
+            locked_node.clone()
+        };
+        registry.store_obj(&drained_node_spec).await?;
```

Comment:
these operations should also go via the operation guard as a kind of update operation,
```rust
NodeOperation::AddDrainingVolumes(DrainingVolumes),
NodeOperation::RemoveDrainingVolumes(DrainingVolumes),
NodeOperation::RemoveAllDrainingVolumes(DrainingVolumes),
NodeOperation::SetDrained,

struct DrainingVolumes {
 volumes: HashSet<VolumeId>,
}
```

Reply:
I suppose that also applies to `set_draining_timestamp_if_none()`

Reply:
fixed

Reply:
Actually, I suppose it doesn't apply to the timestamp, because that never gets persisted.